index.js

```diff
@@ -138,22 +142,27 @@ function commitForTag(gitPath, tag) {
 }
 
 function findTag(gitPath, sha) {
-  var tag = findPackedTag(gitPath, sha);
-  if (tag) { return tag; }
-
-  var tagsPath = path.join(gitPath, 'refs', 'tags');
-  if (!fs.existsSync(tagsPath)) { return false; }
-
-  var tags = fs.readdirSync(tagsPath);
+  var tags = findPackedTags(gitPath, sha)
+    .concat(findUnpackedTags(gitPath, sha));
```

Comment:
The sort order of `readdir` and the `packed-ref` file is not guaranteed. Granted, it is unlikely that either will change but this gives some extra confidence that findTag() will remain deterministic.